library/src/main/scala/za/co/absa/springdocopenapiscala/OpenAPIModelRegistration.scala

```diff
@@ -223,34 +218,63 @@ class OpenAPIModelRegistration(
     }
   }
 
-  private def handleSimpleType(tpe: Type): Schema[_] = {
-    val schema = new Schema
-    val OpenAPISimpleType(terminalTpe, format) = getOpenAPISimpleType(tpe)
-    schema.setType(terminalTpe)
```

Comment:
I would retain the `OpenAPISimpleType` and then have a match case for the `terminalTpe` where you create the concrete schema.
Something like this:
```
terminalTpe match {
  case "integer" => val s = new IntegerSchema()
     format.foreach(f => s.setFormat(f))
    s
       
//etc.
```

 With your impl, you have some duplication, e.g. multiple places with
```
      val s = new IntegerSchema()
      s.setFormat("int32")
```